### PR TITLE
hv: move io_req.h & pci.h

### PR DESCRIPTION
**Pull request by mingqiangchi**
renamed: include/arch/x86/io_req.h -> include/arch/x86/guest/io_req.h
renamed: include/dm/pci.h -> include/hw/pci.h

Tracked-On: #1842
Signed-off-by: Mingqiang Chi <mingqiang.chi@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by JasonChenCJ** [comment]
@dongyaozu seemed we are still not clearly separate io_emul.h and io_req.h, some structure for io_emul are still in io_req.h. @mingqiangchi , let's do a reshuffle again and then we can have include/arch/x86/io_emul.h and include/common/io_req.h.

**Comment by mingqiangchi** [comment]
have updated the patch
renamed:    include/arch/x86/io_req.h -> include/common/io_req.h
renamed:    include/dm/pci.h -> include/hw/pci.h
move some structures from io_req.h to io_emul.h

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/include/arch/x86/guest/io_emul.h [hunk 1]
```diff
@@ -23,6 +23,131 @@
 #define RTC_PIO_IDX		(PM1B_CNT_PIO_IDX + 1U)
 #define EMUL_PIO_IDX_MAX	(RTC_PIO_IDX + 1U)
 
+#define IO_ATTR_R               0U
```
Comment (dongyaozu) [inline hypervisor/include/arch/x86/guest/io_emul.h]: What is the main reason to move these code from common to be X86 specific?  To me, they are common, though they may be used in X86 specific instruction emulation side.

Comment (JasonChenCJ) [reply hypervisor/include/arch/x86/guest/io_emul.h]: the file name in common dir is now io_req.c and it mainly handle io request between DM & guest. Meanwhile, io_emul.c in arch dir is handling io emulation in HV & DM,  so actually io_req.c is a sub-stream of io_emul.c. For these structures, yes they should be common, but it's not suitable to put in io_req.h I think.

**Comment by wenlingz** [comment]
@mingqiangchi please rebase your code, thanks

**Comment by mingqiangchi** [comment]
@wenlingz  need to do some changes for this patch as per eddie's comments, could you hold this path

**Comment by wenlingz** [comment]
> @wenlingz need to do some changes for this patch as per eddie's comments, could you hold this path

OK, got it

**Comment by HuiHuangShi** [comment]
43 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/lib/atomic.h func:atomic_load32 offset:45<br> reason:Static procedure is not explicitly called in code analysed. : atomic_load32<br><2> config:common filename:/hypervisor/include/lib/atomic.h func:atomic_store32 offset:57<br> reason:Static procedure is not explicitly called in code analysed. : atomic_store32<br><3> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:20<br> reason:Procedure is not called or referenced in code analysed. : arch_fire_vhm_interrupt<br><4> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:20<br> reason:Procedure is not called or referenced in code analysed. : arch_fire_vhm_interrupt<br><5> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:30<br> reason:Pointer not checked for null before use. : sos_vm<br><6> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:30<br> reason:Pointer not checked for null before use. : sos_vm<br><7> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_remove_mmio_region offset:35<br> reason:Pointer param should be declared pointer to const. : vm<br><8> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_remove_mmio_region offset:35<br> reason:Pointer param should be declared pointer to const. : pml4_page<br><9> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_remove_mmio_region offset:35<br> reason:Pointer param should be declared pointer to const. : vm<br><10> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_remove_mmio_region offset:35<br> reason:Pointer param should be declared pointer to const. : pml4_page<br><11> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_remove_mmio_region offset:35<br> reason:Procedure is not called or referenced in code analysed. : arch_remove_mmio_region<br><12> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_remove_mmio_region offset:35<br> reason:Procedure is not called or referenced in code analysed. : arch_remove_mmio_region<br><13> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_emulate_mmio_complete offset:47<br> reason:Pointer param should be declared pointer to const. : vcpu<br><14> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_emulate_mmio_complete offset:47<br> reason:Pointer param should be declared pointer to const. : vcpu<br><15> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_emulate_mmio_complete offset:47<br> reason:Procedure is not called or referenced in code analysed. : arch_emulate_mmio_complete<br><16> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_emulate_mmio_complete offset:47<br> reason:Procedure is not called or referenced in code analysed. : arch_emulate_mmio_complete<br><17> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:66<br> reason:Procedure is not called or referenced in code analysed. : emulate_pio_complete<br><18> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:66<br> reason:Procedure is not called or referenced in code analysed. : emulate_pio_complete<br><19> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:67<br> reason:Pointer param should be declared pointer to const. : vcpu<br><20> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:67<br> reason:Pointer param should be declared pointer to const. : vcpu<br><21> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:70<br> reason:Expression needs brackets.<br><22> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:87<br> reason:Pointer param should be declared pointer to const. : vcpu<br><23> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:87<br> reason:Pointer param should be declared pointer to const. : vcpu<br><24> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:87<br> reason:Procedure is not called or referenced in code analysed. : pio_instr_vmexit_handler<br><25> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:87<br> reason:Procedure is not called or referenced in code analysed. : pio_instr_vmexit_handler<br><26> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:120<br> reason:Pointer param should be declared pointer to const. : vcpu<br><27> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:120<br> reason:Pointer param should be declared pointer to const. : vcpu<br><28> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:120<br> reason:Procedure is not called or referenced in code analysed. : ept_violation_vmexit_handler<br><29> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:120<br> reason:Procedure is not called or referenced in code analysed. : ept_violation_vmexit_handler<br><30> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:205<br> reason:Pointer param should be declared pointer to const. : vm<br><31> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:205<br> reason:Pointer param should be declared pointer to const. : vm<br><32> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:205<br> reason:Procedure is not called or referenced in code analysed. : allow_guest_pio_access<br><33> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:205<br> reason:Procedure is not called or referenced in code analysed. : allow_guest_pio_access<br><34> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:212<br> reason:Casting operation to a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br>.....<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by mingqiangchi** [comment]
have updated the patches

**Comment by HuiHuangShi** [comment]
35 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/lib/atomic.h func:atomic_load32 offset:45<br> reason:Static procedure is not explicitly called in code analysed. : atomic_load32<br><2> config:common filename:/hypervisor/include/lib/atomic.h func:atomic_store32 offset:57<br> reason:Static procedure is not explicitly called in code analysed. : atomic_store32<br><3> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:20<br> reason:Procedure is not called or referenced in code analysed. : arch_fire_vhm_interrupt<br><4> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:20<br> reason:Procedure is not called or referenced in code analysed. : arch_fire_vhm_interrupt<br><5> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:30<br> reason:Pointer not checked for null before use. : sos_vm<br><6> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:30<br> reason:Pointer not checked for null before use. : sos_vm<br><7> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:44<br> reason:Procedure is not called or referenced in code analysed. : emulate_pio_complete<br><8> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:44<br> reason:Procedure is not called or referenced in code analysed. : emulate_pio_complete<br><9> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:45<br> reason:Pointer param should be declared pointer to const. : vcpu<br><10> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:45<br> reason:Pointer param should be declared pointer to const. : vcpu<br><11> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:48<br> reason:Expression needs brackets.<br><12> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Pointer param should be declared pointer to const. : vcpu<br><13> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Pointer param should be declared pointer to const. : vcpu<br><14> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Procedure is not called or referenced in code analysed. : pio_instr_vmexit_handler<br><15> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Procedure is not called or referenced in code analysed. : pio_instr_vmexit_handler<br><16> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Pointer param should be declared pointer to const. : vcpu<br><17> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Pointer param should be declared pointer to const. : vcpu<br><18> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Procedure is not called or referenced in code analysed. : ept_violation_vmexit_handler<br><19> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Procedure is not called or referenced in code analysed. : ept_violation_vmexit_handler<br><20> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Pointer param should be declared pointer to const. : vm<br><21> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Pointer param should be declared pointer to const. : vm<br><22> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Procedure is not called or referenced in code analysed. : allow_guest_pio_access<br><23> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Procedure is not called or referenced in code analysed. : allow_guest_pio_access<br><24> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:190<br> reason:Casting operation to a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><25> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:190<br> reason:Casting operation on a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><26> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:192<br> reason:Declaration does not specify an array.       : b<br><27> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Pointer param should be declared pointer to const. : vm<br><28> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Pointer param should be declared pointer to const. : vm<br><29> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Procedure is not called or referenced in code analysed. : deny_guest_pio_access<br><30> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Procedure is not called or referenced in code analysed. : deny_guest_pio_access<br><31> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:204<br> reason:Casting operation to a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><32> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:204<br> reason:Casting operation on a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><33> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:206<br> reason:Declaration does not specify an array.       : b<br><34> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_vm_intr_monitor offset:1175<br> reason:Function return value potentially unused. : (target_vm)->intr_inject_delay_delta<br>.....<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by mingqiangchi** [comment]
fixed violation "Expression needs brackets."

**Comment by mingqiangchi** [comment]
@wenlingz  have updated the patch, could you remove the hold tag?
@dongyaozu  could you review this patch?

**Comment by HuiHuangShi** [comment]
34 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/lib/atomic.h func:atomic_load32 offset:45<br> reason:Static procedure is not explicitly called in code analysed. : atomic_load32<br><2> config:common filename:/hypervisor/include/lib/atomic.h func:atomic_store32 offset:57<br> reason:Static procedure is not explicitly called in code analysed. : atomic_store32<br><3> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:20<br> reason:Procedure is not called or referenced in code analysed. : arch_fire_vhm_interrupt<br><4> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:20<br> reason:Procedure is not called or referenced in code analysed. : arch_fire_vhm_interrupt<br><5> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:30<br> reason:Pointer not checked for null before use. : sos_vm<br><6> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:arch_fire_vhm_interrupt offset:30<br> reason:Pointer not checked for null before use. : sos_vm<br><7> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:44<br> reason:Procedure is not called or referenced in code analysed. : emulate_pio_complete<br><8> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:44<br> reason:Procedure is not called or referenced in code analysed. : emulate_pio_complete<br><9> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:45<br> reason:Pointer param should be declared pointer to const. : vcpu<br><10> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:emulate_pio_complete offset:45<br> reason:Pointer param should be declared pointer to const. : vcpu<br><11> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Pointer param should be declared pointer to const. : vcpu<br><12> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Pointer param should be declared pointer to const. : vcpu<br><13> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Procedure is not called or referenced in code analysed. : pio_instr_vmexit_handler<br><14> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:pio_instr_vmexit_handler offset:65<br> reason:Procedure is not called or referenced in code analysed. : pio_instr_vmexit_handler<br><15> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Pointer param should be declared pointer to const. : vcpu<br><16> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Pointer param should be declared pointer to const. : vcpu<br><17> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Procedure is not called or referenced in code analysed. : ept_violation_vmexit_handler<br><18> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:ept_violation_vmexit_handler offset:98<br> reason:Procedure is not called or referenced in code analysed. : ept_violation_vmexit_handler<br><19> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Pointer param should be declared pointer to const. : vm<br><20> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Pointer param should be declared pointer to const. : vm<br><21> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Procedure is not called or referenced in code analysed. : allow_guest_pio_access<br><22> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:183<br> reason:Procedure is not called or referenced in code analysed. : allow_guest_pio_access<br><23> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:190<br> reason:Casting operation to a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><24> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:190<br> reason:Casting operation on a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><25> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:allow_guest_pio_access offset:192<br> reason:Declaration does not specify an array.       : b<br><26> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Pointer param should be declared pointer to const. : vm<br><27> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Pointer param should be declared pointer to const. : vm<br><28> config:part_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Procedure is not called or referenced in code analysed. : deny_guest_pio_access<br><29> config:sharing_config filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:197<br> reason:Procedure is not called or referenced in code analysed. : deny_guest_pio_access<br><30> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:204<br> reason:Casting operation to a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><31> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:204<br> reason:Casting operation on a pointer. : (unsigned char to unsigned int*): ( uint32_t * ) vm -> arch_vm.io_bitmap<br><32> config:common filename:/hypervisor/arch/x86/guest/vmx_io.c func:deny_guest_pio_access offset:206<br> reason:Declaration does not specify an array.       : b<br><33> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_vm_intr_monitor offset:1175<br> reason:Function return value potentially unused. : (target_vm)->intr_inject_delay_delta<br><34> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_vm_intr_monitor offset:1175<br> reason:Function return value potentially unused. : (target_vm)->intr_inject_delay_delta<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**